ladybug/src/main/java/nl/nn/ibistesttool/IbisDebuggerAdvice.java

```diff
@@ -189,58 +177,66 @@ public PipeRunResult debugPipeGetInputFrom(ProceedingJoinPoint proceedingJoinPoi
 			try {
 				Object[] args = proceedingJoinPoint.getArgs();
 				args[messageParamIndex] = message;
-				result = (M)proceedingJoinPoint.proceed(args);
+				result = proceedingJoinPoint.proceed(args);
 			} catch(Throwable throwable) {
 				throw ibisDebugger.senderAbort(sender, messageId, throwable);
-			} finally {
-				if (capture!=null && ibisDebugger instanceof Debugger) {
-					((Debugger)ibisDebugger).capturedInput(messageId, capture.toString());
-				}
 			}
 		} else {
 			// Resolve parameters so they will be added to the report like when the sender was not stubbed and would
 			// resolve parameters itself
 			if (sender instanceof IWithParameters) {
 				ParameterList parameterList = ((IWithParameters)sender).getParameterList();
 				if (parameterList!=null) {
-					parameterList.getValues(message, session);
+					parameterList.getValues((Message)message, session);
 				}
 			}
-			if (capture!=null && ibisDebugger instanceof Debugger) {
-				((Debugger)ibisDebugger).capturedInput(messageId, capture.toString());
-			}
 		}
 		if (sender instanceof SenderWrapperBase && ((SenderWrapperBase)sender).isPreserveInput()) {
 			// signal in the debugger that the result of the sender has been replaced with the original input
-			result = (M)ibisDebugger.preserveInput(messageId, (Message)result);
-		}
-		if (result!=null && result instanceof PipeRunResult) {
-			PipeRunResult prr = (PipeRunResult)result;
-			prr.setResult(ibisDebugger.senderOutput(sender, messageId, prr.getResult()));
-			return result;
+			result = ibisDebugger.preserveInput(messageId, (Message)result);
 		}
-		return (M)ibisDebugger.senderOutput(sender, messageId, Message.asMessage(result));
+		return result;
 	}
 	
 	/**
 	 * Provides advice for {@link ISender#sendMessage(Message message, IPipeLineSession session)}
 	 */
 	public Message debugSenderInputOutputAbort(ProceedingJoinPoint proceedingJoinPoint, Message message, IPipeLineSession session) throws Throwable {
-		return (Message)debugSenderInputOutputAbort(proceedingJoinPoint, message, session, 0);
+		Object result = debugSenderInputStubAbort(proceedingJoinPoint, message, session, 0);
+		ISender sender = (ISender)proceedingJoinPoint.getTarget();
+		String messageId = session == null ? null : session.getMessageId();
+		return ibisDebugger.senderOutput(sender, messageId, Message.asMessage(result));
```

Comment:
Dit riep debugSenderInputOutputAbort(). Is dat niet meer nodig?

Reply:
Van de debugSenderInputOutputAbort() heb ik het Output stuk logica verwijderd en daarom de methode hernoemd naar debugSenderInputStubAbort() die nog steeds aangeroepen wordt. Het Output stuk zit nu in de verschillende methodes zelf omdat de ene methode met Message werkt en de ander met PipeRunResult.

Reply:
De drie publieke debugSenderInputOutputAbort() methods waren met opzet éénregelige methods() die al het werk door dezelfde implementatie lieten doen. Dat wil ik zo houden.
Je kan wel bijvoorbeeld een parameter ``boolean expectPipeRunResult`` toevoegen, en op grond daarvan de extra code uitvoeren.